ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,23 +1,34 @@
 """Serialization Strategy."""
 
-from typing import Any, Callable
+from typing import Any, Callable, Dict
 
 from google.protobuf import type_pb2
-from google.protobuf.internal import decoder
-from google.protobuf.internal import encoder
+from google.protobuf.descriptor import FieldDescriptor
+from google.protobuf.internal import decoder, encoder
+from typing_extensions import Buffer
 
+from ni_measurementlink_service._internal.parameter import _message
+from ni_measurementlink_service._internal.stubs.ni.protobuf.types import xydata_pb2
 
-def _scalar_encoder(encoder) -> Callable[[int], Callable]:
+LowLevelEncoder = Callable[[Callable[[Buffer], int], Any, None], None]
+EncoderFactory = Callable[[int], LowLevelEncoder]
+LowLevelDecoder = Callable[
+    [memoryview, int, int, type_pb2.Field.Kind.ValueType, Dict[FieldDescriptor, Any]], int
+]
+DecoderFactory = Callable[[int, str], LowLevelDecoder]
+
+
+def _scalar_encoder(encoder) -> EncoderFactory:
```

Comment:
This file imports `encoder` and `decoder` and then shadows them with parameters. We should rename one or the other:
- We could `import google.protobuf.internal.decoder as _decoder`
- Or we could rename the parameter to something like `inner_decoder` or `decoder_factory` (if that's what it is; it's the same as our `DecoderFactory` with more parameters)

Reply:
Do you still feel we want to rename this to, e.g. decoder_constructor, or is it clear enough with the type hint next to it? I think it's clear as is.